radicle-httpd/src/api/json.rs

```diff
@@ -46,14 +47,22 @@ pub(crate) fn session(session_id: String, session: &Session) -> Value {
 }
 
 /// Returns JSON for a blob with a given `path`.
-pub(crate) fn blob(blob: &Blob, path: &str) -> Value {
-    json!({
+pub(crate) fn blob<T: AsRef<[u8]>>(blob: &Blob<T>, path: &str) -> Value {
+    let mut response = json!({
         "binary": blob.is_binary(),
-        "content": blob.content(),
         "name": name_in_path(path),
         "path": path,
         "lastCommit": commit(blob.commit())
-    })
+    });
+
+    if !blob.is_binary() {
+        match str::from_utf8(blob.content()) {
+            Ok(content) => response["content"] = content.into(),
+            Err(err) => return json!({"success": false, "error": err.to_string()}),
```

Comment:
```suggestion
            Err(err) => return json!({ "error": err.to_string() }),
```
`success` should only be used for PUT/POST/PATCH, ie. if there's an operation being carried out.
Looks good otherwise!

Reply:
[diff](https://github.com/radicle-dev/heartwood/compare/82a8189ae11c6a6612fc3bf4ba3cd92928f1494e..69bf6421accc8a349cd36fe79d5c7e5479ebf647)